File: kotlin-spring-boot-base/src/main/java/com/xiao/boot/base/property/EnvProperty.java
package com.xiao.boot.base.property;

import com.xiao.boot.base.ServerConstants;
import com.xiao.boot.base.env.ProfileType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Repeatable(EnvProperties.class)
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface EnvProperty {
    ProfileType[] profiles() default {};
    String value() default "";
    boolean allowEmpty() default false;
    boolean encrypt() default false;
    String encryptKey() default ServerConstants.ENV_ENCRYPT_KEY;
}

File: kotlin-spring-boot-base/src/main/kotlin/com/xiao/boot/base/property/EnvProperty.kt
package com.xiao.boot.base.property

import com.xiao.boot.base.env.EnvConstants.ENV_ENCRYPT_KEY
import com.xiao.boot.base.env.ProfileType

/**
 * Target class fileds must be public.
 *
 * @author lix wang
 */
@Repeatable
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class EnvProperty(
    val profiles: Array<ProfileType> = [],
    val value: String = "",
    val allowEmpty: Boolean = false,
    val encrypt: Boolean = false,
    val encryptKey: String = ENV_ENCRYPT_KEY
)